packages/picker/test/index.ts

```diff
@@ -1412,6 +1412,41 @@ export function runPickerTests(): void {
                 'has focus ring?'
             ).to.be.false;
         });
+        it('does not close on document scroll', async () => {
+            const el = await fixture(html`
+                <div style="height: 200vh; padding: 50vh 0;">
+                    <sp-picker label="Select an option" placement="right">
+                        <sp-menu-item value="option-1">Option 1</sp-menu-item>
+                        <sp-menu-item value="option-2">Option 2</sp-menu-item>
+                        <sp-menu-item value="option-3">Option 3</sp-menu-item>
+                    </sp-picker>
+                </div>
+            `);
+
+            const picker = el.querySelector('sp-picker') as Picker;
+            await elementUpdated(picker);
+            await nextFrame();
+            await nextFrame();
+
+            expect(picker.open).to.be.false;
+
+            const opened = oneEvent(picker, 'sp-opened');
+            picker.click();
+            await opened;
+
+            expect(picker.open).to.be.true;
+
+            // Scroll the document
+            if (document.scrollingElement) {
+                document.scrollingElement.scrollTop = 100;
```

Comment:
I'm wondering if the tooling has any event hooks we can connect to that indicate when the scroll action is completed? Might let us reduce the need for the arbitrary awaits afterwards.

Reply:
I want us to avoid adding any new tests that would add to our existing list of flaky tests, which is why I raise this as a concern because generic await times don't guarantee the test is ready to be performed.

Reply:
calling out the elementUpdated(picker) here too, nextFrame() is the fallback but most cases should work with elementUpdated

Reply:
Thanks. A little bit of context, when multiple overlays or components uses `nextFrame()` simultaneously, the timing of their resolutions might not be predictable, especially during scroll events or rapid state changes. This is where most of the flakiness happens in the virtual triggers. In this context however this is a linear event and I think we can use the `nextFrame()` to safely add the resolution but on the safer side we can definitely and should go with a more robust approach with `waitUntil` like you stated.